### PR TITLE
h264: wrote readUes function and using it for series of ue parses

### LINKED ISSUE #45: h264: readUes function to reduce repetitiveness of code
https://github.com/ausocean/h264decode/blob/3465ea9616833cc1b620f61ce15353ee733cecc0/h264/sps.go#L413-L431
Note the repetitiveness of this code. We should have a func for the parsing of multiple Ue fields in a row.

### PR DESCRIPTION
**Pull request by saxon-milton**
readUes can be used where there is a series of parses of ue fields. It has been used in code where this occurs.

Resolves #45

### CODE REVIEW

File: h264/read.go [hunk 1]
```diff
@@ -4,6 +4,9 @@ import (
 	"fmt"
 	"io"
 	"os"
+
+	"github.com/ausocean/h264decode/h264/bits"
+	"github.com/pkg/errors"
```
Comment (kortschak) [inline h264/read.go:9]: See previous comment.

File: h264/read.go [hunk 2]
```diff
@@ -301,3 +304,20 @@ func (b *BitReader) StreamPosition() (int, int, int) {
 func (b *BitReader) LogStreamPosition() {
 	logger.Printf("debug: %d byte stream @ byte %d bit %d\n", len(b.bytes), b.byteOffset, b.bitOffset)
 }
+
+type ue struct {
+	loc  *int
```
Comment (kortschak) [inline h264/read.go:309]: Why use a pointer here?

Comment (saxon-milton) [reply h264/read.go:309]: so I can give the fields as args and set them.

Comment (kortschak) [reply h264/read.go:309]: That's not a nice way to do things.

Comment (saxon-milton) [reply h264/read.go:309]: Then what's a nice way to do things ? Please give alternatives with criticisms.

Comment (kortschak) [reply h264/read.go:309]: If you are mutating a value, return the mutated value.

`func mutator(v ue) ue`

Comment (saxon-milton) [reply h264/read.go:309]: see comment sps.go:534

File: h264/sps.go [hunk 3]
```diff
@@ -531,34 +515,17 @@ func NewSPS(rbsp []byte, showPacket bool) (*SPS, error) {
 			if v := b.NextField("MotionVectorsOverPicBoundaries", 1); v == 1 {
 				sps.MotionVectorsOverPicBoundaries = true
 			}
-			sps.MaxBytesPerPicDenom, err = readUe(nil)
```
Comment (saxon-milton) [inline h264/sps.go:534]: @kortschak I'm trying to reduce the verbosity of a series of readUe calls. See the change I made here using readUes. I need to give references to the fields, or is this not worth it ?

**Pull request closed without merging**